source/contribute.rst

```diff
@@ -166,7 +166,36 @@ Now do some work, make some changes then commit them:
 .. code-block:: console
 
     $ git status
-    $ git commit <filespec>
+    $ git add -p
+    $ git commit -v
+
+Commit messages should look like emails, meaning they should have a
+subject, but also a body. The subject should be on the first line, and
+not exceed 50 chars. It should tell us what you did, and every change in
+the diff should have to do with that subject. The body should be
+separated from it by a blank line and should tell us *why* you did what
+you did. That is also a good place to tell people about alternate
+solutions that were considered and the reasons for rejecting them. Links
+to related issues are more than welcome, but should be summarized so
+that the pull request can be understood without resorting to them.
+Ideally, the git history should be understandable without a network
+connection. Here is an example of a good although fictitious commit
+message::
```

Comment:
```suggestion
message:
```

Reply:
That's not a typo, that's a way to get a preformatted output, you can see it the rich diff

Reply:
I did watch it in a md-editor, but if this will have the same result on the website as in the preformatted output, I'm fine with it.

Reply:
I'm not sure actually, and I don't know how to figure it out, I tried to build the docs locally and failed.

Reply:
Seems fine to me :-)